rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt

```diff
@@ -22,44 +23,50 @@ private fun clear(value: String, type: Type): String {
 fun movel(
     operationExtender: String?,
     target: AssignableExpression,
-    value: Expression,
+    source: Expression,
+    dataAttributes: Expression?,
     interpreterCore: InterpreterCore
 ): Value {
-    if (value !is FigurativeConstantRef) {
-        if (value.type() is ArrayType) {
-            throw UnsupportedOperationException("Cannot set an array as factor 2 in MOVEL/MOVEL(P) statement")
+    if (source is FigurativeConstantRef) {
```

Comment:
Against the colorful differences, was a little refactored because I had to add a new check of source type. I decided to reorder the logic by:
- moving the easy returns at the top instead the end;
- changing inner `if` with a `when` to perform the right operation based of source type.

If you want, I can restore it and add an "else if" for the check of new type.

Reply:
The different code style, as explained also to your colleagues, is allowed just for new functions.